### ISSUE TITLE
Add a menubar

### ISSUE THREAD

**Issue by gyscos** [gyscos/Cursive#27]
Not necessarily a `View` in itself, but at least something that can be added to the root `Cursive`.

Notes:
* Should it be global, or per screen?
* Should it be available when a popup is visible?
* When opening a window (ex: options), it should make sure the window does not already exist (or raise it? Sounds dangerous if a popup from Options is shown...)
* Togglable auto-hide (reserves 1 row when off), configurable "show" button (ex: F10).
    * When auto-hidden and with no "show" button, acts as if no menubar was present.
* Must be able to draw the "popup" menus outside of its regular bounds, so not a traditional `View`...

**Comment by gyscos** [comment]
Current solution: 
* Global menu. May become wipeable if needed.
* Dialogs never block the menubar.
* Window raise is a separate issue and will have to do with the `StackView`.
* Can toggle auto-hide

Concerning the display, should it do everything manually, or use a kind of `SelectView`, if only for the drawing part?...

**Comment by matthiasbeyer** [comment]
From mobile, so markup might be fucked up...

>   •  Should it be global, or per screen?

Let the user decide!

>  •  Should it be available when a popup is visible?

Again: let the user decide (in the popup, not the menubar)!


>  •  When opening a window (ex: options), it should make sure the window does not already exist (or raise it? Sounds dangerous if a popup from Options is shown...)

Not sure what is possible, but maybe one wants to focus a window this way (if there is something like multi-window with focus and so on) 

>  •  Togglable auto-hide (reserves 1 row when off), configurable "show" button (ex: F10). 

Why not redraw the whole screen and use the empty row when the menubar is not shown?

I hope I do not annoy you.. Just want to contribute some discussion!

**Comment by gyscos** [comment]
Popup menus will be additional layers, so they may be regular `View`s after all.

Also, "select" views (drop-down, not like in the example) could use the same kind of view, so it's indeed reusable:
![screenshot from 2016-06-27 21-18-48](https://cloud.githubusercontent.com/assets/564903/16403700/c68d8b5e-3cac-11e6-8931-ad2fc9dc1e08.png)

![screenshot from 2016-06-27 21-17-20](https://cloud.githubusercontent.com/assets/564903/16403683/a061d78c-3cac-11e6-8793-86f9ad39b957.png)

**Comment by gyscos** [comment]
Currently blocked on #36 and #37.

**Comment by gyscos** [comment]
There is now a mostly working `menubar` example:

![screenshot from 2016-07-02 19-39-08](https://cloud.githubusercontent.com/assets/564903/16543507/b10cba0a-408c-11e6-8637-5697918cd114.png)

**Comment by matthiasbeyer** [comment]
Wow, this looks awesome! Thanks a lot for your work on this!
